src/blocks/pomodoro.rs

```diff
@@ -15,76 +17,107 @@ use crate::widget::I3BarWidget;
 use crate::widgets::button::ButtonWidget;
 
 enum State {
-    Started,
+    Started(DateTime<Utc>),
     Stopped,
-    Paused,
-    OnBreak,
+    Paused(Duration),
+    OnBreak(DateTime<Utc>),
+}
+
+impl State {
+    fn elapsed(&self) -> Duration {
+        match self {
+            State::Started(start) => Utc::now()
+                .signed_duration_since(start.to_owned())
+                .to_std()
+                .unwrap_or_default(),
+            State::Stopped => unreachable!(),
+            State::Paused(duration) => duration.to_owned(),
+            State::OnBreak(start) => Utc::now()
+                .signed_duration_since(start.to_owned())
+                .to_std()
+                .unwrap_or_default(),
+        }
+    }
+}
+
+impl fmt::Display for State {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            State::Stopped => write!(f, "\u{25a0} 0:00"),
+            State::Started(_) => write!(
+                f,
+                "\u{f04b} {}:{:02}",
+                self.elapsed().as_secs() / 60,
+                self.elapsed().as_secs() % 60
+            ),
+            State::OnBreak(_) => write!(
+                f,
+                "\u{2615} {}:{:02}",
+                self.elapsed().as_secs() / 60,
+                self.elapsed().as_secs() % 60
+            ),
+            State::Paused(duration) => write!(
+                f,
+                "\u{2615} {}:{:02}",
```

Comment:
I would have kept this to the pause icon `\u{f04c}`

Reply:
Had copied the wrong one. fixed